praw/models/reddit/collections.py

```diff
@@ -459,11 +477,17 @@ def __iter__(self):
                print(collection.permalink)
 
         """
-        request = self._reddit.get(
+        response_data = self._reddit.request(
+            "GET",
             API_PATH["collection_subreddit"],
             params={"sr_fullname": self.subreddit.fullname},
         )
-        for collection in request:
+        self._reddit._objector.check_error(response_data)
```

Comment:
It seems like nearly every time we check for errors. Can we do that implicitly in `request`? I'm guessing there are probably a few times we don't want to check for errors. Perhaps we can add a `request_without_error_check` method for those few cases so we needn't add `check_error` calls in all these methods.

Reply:
That sounds like a good idea. To avoid making too many potentially high impact changes though I’ve made a `_request_and_check_error()` for now. This new function should be applied to endpoints that are known to have an `errors` key in the JSON data.

I’ve used regular `request()` on endpoints that don’t return a dict. This could be risky though because I don’t know for sure if it’s possible for any of these endpoints to decide to change format and return a dict when an error occurs. There aren’t many tests that test API errors, and it’s not clear why some methods choose to use `.request()` (which doesn’t error check) over `.get()`/`.post()`, etc.